src/App/Formatter/ExportJob.php

```diff
@@ -12,8 +12,42 @@
 namespace Ushahidi\App\Formatter;
 
 use Ushahidi\Core\Traits\FormatterAuthorizerMetadata;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Carbon;
+use Carbon\CarbonInterval;
 
 class ExportJob extends API
 {
     use FormatterAuthorizerMetadata;
+
+    protected function formatUrl($value)
+    {
+        if (empty($value)) {
+            return $value;
+        }
+
+        // If we already have a URL, just return it
+        if (filter_var($value, FILTER_VALIDATE_URL)) {
+            return $value;
+        }
+
+        // Removes path from image file name, encodes the filename, and joins the path and filename together
+        $url_path = explode("/", $value);
+        $filename = rawurlencode(array_pop($url_path));
+        array_push($url_path, $filename);
+        $path = implode("/", $url_path);
+
+        $expiration = Carbon::now()->add(CarbonInterval::fromString(config('media.temp_url_lifespan')));
+
+        // Try to get a temporary URL
+        try {
+            return Storage::temporaryUrl($path, $expiration);
```

Comment:
@rowasc @kinstelli I think this does this job? Although it'll generate a new URL each time